unicorn/include/cocaine/zookeeper/handler.hpp

```diff
@@ -0,0 +1,230 @@
+/*
+* 2015+ Copyright (c) Anton Matveenko <antmat@yandex-team.ru>
+* All rights reserved.
+*
+* This program is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*/
+#ifndef ZOOKEEPER_HANDLER_HPP
+#define ZOOKEEPER_HANDLER_HPP
+
+#include "cocaine/zookeeper/zookeeper.hpp"
+#include "cocaine/zookeeper/exception.hpp"
+
+#include <cocaine/locked_ptr.hpp>
+
+#include <functional>
+#include <string>
+#include <memory>
+#include <vector>
+
+#include <unordered_map>
+#include <cassert>
+
+namespace zookeeper {
+
+typedef Stat node_stat;
+
+//Used to prevent direct creation of managed handlers derived classes.
+class handler_tag {
```

Comment:
It's for different reason. This tag prevents any construction of handler derived class but from handler_scope_t. This prevents misusage of derived classes in zookeeper::connection